pandas/tests/groupby/test_groupby.py

```diff
@@ -2846,3 +2846,22 @@ def test_sum_of_booleans(n):
     result = df.groupby("groupby_col").sum()
     expected = DataFrame({"bool": [n]}, index=Index([1], name="groupby_col"))
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("method", ["head", "tail", "nth", "first", "last"])
+def test_groupby_method_drop_na(method):
+    # GH 21755
+    df = DataFrame({"A": ["a", np.nan, "b", np.nan, "c"], "B": range(5)})
+
+    if method == "nth":
+        result = df.groupby("A").agg(method, n=0)
```

Comment:
Could you call the actual methods instead of using `agg`?

Reply:
Okay, how about this time?